### PR TITLE
Use fill=`Inf` to sample full file for right fill guess

### LINKED ISSUE #2727: Do not suggests fread fill=TRUE if already used
current fread behavior.
```r
dt=fread("Rprofmem.out", header=FALSE)
#Warning message:
#In fread("Rprofmem.out", header = FALSE) :
#  Stopped early on line 2. Expected 3 fields but found 2. Consider fill=TRUE and comment.char=. First #discarded non-empty line: <<new page:"library" >>
dt=fread("Rprofmem.out", header=FALSE, fill=TRUE)
#Warning message:
#In fread("Rprofmem.out", header = FALSE, fill = TRUE) :
#  Stopped early on line 90. Expected 7 fields but found 8. Consider fill=TRUE and comment.char=. First #discarded non-empty line: <<new page:"close" "readRDS" "FUN" "lapply" "find.package" "system.file" #"library" >>
```
Before printing this warning we should check if `fill=TRUE` was used.

### PR DESCRIPTION
**Pull request by ben-schwen**
Closes #2727 
Follow-up to #5119 

- [x] fix
- [x] doc
- [x] news
- [x] tests

### CODE REVIEW

File: R/fread.R [hunk 1]
```diff
@@ -28,7 +28,7 @@ yaml=FALSE, autostart=NA, tmpdir=tempdir(), tz="UTC")
     isTRUEorFALSE(stringsAsFactors) || (is.double(stringsAsFactors) && length(stringsAsFactors)==1L && 0.0<=stringsAsFactors && stringsAsFactors<=1.0),
     is.numeric(nrows), length(nrows)==1L
   )
-  fill=as.integer(fill)
+  fill = if(is.infinite(fill)) .Machine$integer.max else as.integer(fill)
```
Comment (tdhock) [inline R/fread.R:31]: maybe identical(fill, Inf) instead? 
is.infinite(-Inf) is TRUE

**Review by tdhock — CHANGES_REQUESTED** [review]

File: R/fread.R (hunk 1 above)
Comment (ben-schwen) [reply R/fread.R:31]: I had the same thought when writing it but thought if someone if willing to `fill=-Inf` then he deserves it :D

But yeah, should be unambigous in light of the mutation testing

Comment (tdhock) [reply R/fread.R]: ok great thx

**Review by tdhock — APPROVED** [review]

**Comment by tdhock** [comment]
looks good to me, please merge

**Pull request merged**